### DISCUSSION TITLE
Query timestampz via chrono:DateTime

### DISCUSSION THREAD

**Discussion by 1g0rkR** [diesel-rs/diesel#3472] [Q&A] [answered]
Hi folks,

I would appreciate if somebody could support me on querying Timestampz via chrono crate. I wrote such structure:
```
#[derive(Queryable)]
pub struct GetPendingOrgDao {
    pub id:         Uuid,
    pub name:       String,
    pub created_at: DateTime,
    pub updated_at: DateTime,
}
```
but DateTime requires to specify the TimeZone.  How could I read timestamp with time zone from DB? Time zones can be different.
Next question, What I will have in the result using NaiveDateTime? just date/time without time zone? 

Thanks,
Igor

**Comment by weiznich** [comment, marked as answer]
It's impossible to answer such questions without more context. Please include at least the following information:
* Your diesel version including all features you've enabled
* Your rustc version
* The relevant part of your schema.rs file
* The corresponding query function
* Potentially the error message generated by rustc

> but DateTime requires to specify the TimeZone. How could I read timestamp with time zone from DB? Time zones can be different.

Yes you need to specify a target timezone, as the database does not actually store any timezone information at all. It just normalizes all timestamps to be in the same timezone and does a transformation to load it into the requested target timezone.

> Next question, What I will have in the result using NaiveDateTime? just date/time without time zone?

That returns date/time as UTC time as it is stored by postgresql

**Reply by 1g0rkR** [reply]
Thanks Weiznich,

You answered to my question by postgres logic for timestampz.
